File: jscomp/test/loop_regression_test.js
'use strict';

var Mt = require("./mt.js");

function f(param) {
  var v = {
    contents: 0
  };
  var acc = {
    contents: 0
  };
  var n = 10;
  while(true) {
    if (v.contents > n) {
      return acc.contents;
    }
    acc.contents = acc.contents + v.contents | 0;
    v.contents = v.contents + 1 | 0;
    continue ;
  };
}

var suites_0 = [
  "sum",
  (function (param) {
      return {
              tag: /* Eq */0,
              _0: 55,
              _1: f(undefined)
            };
    })
];

var suites = /* :: */{
  _0: suites_0,
  _1: /* [] */0
};

Mt.from_pair_suites("Loop_regression_test", suites);

exports.f = f;
exports.suites = suites;
/*  Not a pure module */

File: jscomp/test/loop_regression_test.ml
let f () =
  let v = ref 0 in
  let acc = ref 0 in
  let rec loop n : int =
    if !v > n then !acc
    else (
      acc := !acc + !v ;
      incr v ;
      loop n ) in
  loop 10

let suites = Mt.[("sum", fun _ -> Eq (55, f ()))]

;;
Mt.from_pair_suites __MODULE__ suites
